### ISSUE TITLE
Feature? / Bug?: User friendly preload engine migration

### ISSUE THREAD

**Issue by ftake** [ibus/ibus#2274]
In IBus 1.5.23, some xkb engines including xkb:de::ger were renamed. After upgrade, IBus daemon cannot load an engine with the old name specified in preload engines of configuration DB. So it may load en layout engine as a fallback (if the preload engine list becomes empty).

This behavior is very confusing for users who does not know IBus is controlling keyboard layouts.
For example: https://bugzilla.opensuse.org/show_bug.cgi?id=1177545

I'm wondering if more user-friendly migration is possible like:

- Keep old names in simple.xml and migrate configuration file automatically
- Show a message that an unknown engine was specified in the configuration
- Guess current keyboard layout and add it to preload engine instead of xkb:us::eng

**Comment by fujiwarat** [comment]
It might be a regression in 1.5.23 when I changed to generate simple.xml from a whitelist to a blacklist.
Now simple.xml is generated from the latest xkeyboard-config by IBus release to retrieve the necessary keymaps.
xkeyboard-config has a language "deu" for "de" layout. I don't remember whether it was updated some years ago or the previous whilelist was not generated from it.

> Keep old names in simple.xml and migrate configuration file automatically

Probably I don't wish to keep the hard-coded exception.

> Show a message that an unknown engine was specified in the configuration

I show a warning dialog when all engines are unknown at the moment. I will think the request again later.

> Guess current keyboard layout and add it to preload engine instead of `xkb:us::eng`.

Probably I don't wish to have the hard-coded conversion from unknown engines and I don't know how many engines are unknown in 1.5.23.
